srcopsmetrics/github_knowledge.py

```diff
@@ -143,21 +169,12 @@ def get_referenced_issues(pull_request: PullRequest) -> List[int]:
         """
         issues_referenced = []
         for comment in pull_request.get_issue_comments():
-            message = comment.body.split(" ")
-            for idx, word in enumerate(message):
-                if word.replace(":", "") in ISSUE_KEYWORDS:
-                    try:
-                        _LOGGER.info("      ...found keyword referencing issue")
-                        referenced_issue_number = message[idx + 1]
-                        assert (referenced_issue_number).startswith("https")
-                        # last element of url is always the issue number
-                        issues_referenced.append(referenced_issue_number.split("/")[-1])
-                        _LOGGER.info("      ...referenced issue number present")
-                        # we assure that this was really referenced issue
-                        # and not just a keyword without number
-                    except (IndexError, AssertionError) as e:
-                        _LOGGER.info("      ...referenced issue number absent")
-                        _LOGGER.debug(str(e))
+            for id in GitHubKnowledge.search_for_references(comment.body):
+                issues_referenced.append(id)
+
+        for id in GitHubKnowledge.search_for_references(pull_request.body):
+            issues_referenced.append(id)
+
```

Comment:
Use `self`.method_name() and remove static method from `get_referenced_issues` because the method is no more static. Therefore `def get_referenced_issues(self, pull_request: PullRequest) -> List[int]`

Reply:
but the method ```get_referenced_issues``` does not use ```self``` within itself, therefore it should be static, no?

Reply:
well that method inside it's using methods of the class, therefore it requires `self`.

Reply:
right, fixed